balboa-agent/src/main/scala/com/socrata/balboa/agent/LoggingScheduledExecutor.scala

```diff
@@ -0,0 +1,26 @@
+package com.socrata.balboa.agent
+
+import java.util.concurrent.{ScheduledFuture, TimeUnit, ScheduledThreadPoolExecutor}
+
+import com.typesafe.scalalogging.slf4j.Logger
+import org.slf4j.LoggerFactory
+
+class LoggingScheduledExecutor(poolSize: Int) extends ScheduledThreadPoolExecutor(poolSize) {
+  override def scheduleAtFixedRate(runnable: Runnable, delay: Long, period: Long, unit: TimeUnit): ScheduledFuture[_] = {
+    val wrapped = new LogRunnableException(runnable)
+    super.scheduleAtFixedRate(wrapped, delay, period, unit)
+  }
+
+  private class LogRunnableException(underlying: Runnable) extends Runnable {
+    val logger: Logger = Logger(LoggerFactory getLogger getClass.getName)
+
+    override def run(): Unit = {
+      try {
+        underlying.run()
+      } catch {
+        case e: Exception =>
+          logger.error("Underlying runnable threw an exception", e)
+      }
+    }
+  }
+}
```

Comment:
Why are you building a new class? Why isn't it sufficient to add a catch clause to the runnable in BalboaAgent?